### ISSUE TITLE
ContractAction missing resultData

### ISSUE THREAD

**Issue by steven-sheehy** [hashgraph/hedera-services#9322]
### Description

Seeing an issue on mainnet where contract actions do not have a valid resultData for CONTRACT_REVERT_EXECUTED. This results in a warning log in the mirror node:
2023-10-17T12:05:05.012Z ERROR scheduling-3 c.h.m.i.d.ContractResultServiceImpl Recoverable error. Invalid result data for contract action at 1697544296046643003: RESULTDATA_NOT_SET

TransactionBody:
```
transactionID {
  transactionValidStart {
    seconds: 1697544282
    nanos: 178548931
  }
  accountID {
    accountNum: 3022466
  }
}
nodeAccountID {
  accountNum: 21
}
transactionFee: 200000000
transactionValidDuration {
  seconds: 120
}
memo: "HashPack swap HELI to HBAR"
contractCall {
  contractID {
    contractNum: 3045981
  }
  gas: 1120000
  functionParameters: "\030\313\257\345\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000H\313\220\037\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\006\225A\310\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\240\000\000\000\000\000\000\000\000\000\000\000\000\250\266\264y\357)@\242\v\312\211\270\201D\031\261\2444\351\235\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\001\213=\230\203b\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\002\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\035\220\311\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\026;Z"
}
```

TransactionRecord:
```
receipt {
  status: CONTRACT_REVERT_EXECUTED
  contractID {
    contractNum: 3045981
  }
  exchangeRate {
    currentRate {
      hbarEquiv: 30000
      centEquiv: 140529
      expirationTime {
        seconds: 1697547600
      }
    }
    nextRate {
      hbarEquiv: 30000
      centEquiv: 140494
      expirationTime {
        seconds: 1697551200
      }
    }
  }
}
transactionHash: "\203a\257)\246,\337\3425\266\247\337\331\274R\334_V\250\316\377\312jM%@Upl\206,\314\331\200z\032\330\343\377\304\023\332\vx\035\254\t\035"
consensusTimestamp {
  seconds: 1697544296
  nanos: 46643003
}
transactionID {
  transactionValidStart {
    seconds: 1697544282
    nanos: 178548931
  }
  accountID {
    accountNum: 3022466
  }
}
memo: "HashPack swap HELI to HBAR"
transactionFee: 162176000
contractCallResult {
  errorMessage: "0x"
  gasUsed: 896000
}
transferList {
  accountAmounts {
    accountID {
      accountNum: 98
    }
    amount: 162176000
  }
  accountAmounts {
    accountID {
      accountNum: 3022466
    }
    amount: -162176000
  }
}
```

ContractAction:
```
call_type: CALL
calling_account {
  accountNum: 3022466
}
gas: 1099000
input: "\030\313\257\345\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000H\313\220\037\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\006\225A\310\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\240\000\000\000\000\000\000\000\000\000\000\000\000\250\266\264y\357)@\242\v\312\211\270\201D\031\261\2444\351\235\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\001\213=\230\203b\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\002\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\035\220\311\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\026;Z"
recipient_contract {
  contractNum: 3045981
}
call_operation_type: OP_CALL
```

### Steps to reproduce

Execute a contract call that causes a `CONTRACT_REVERT_EXECUTED` response code.

### Hedera network

mainnet

### Version

0.41.x

### Operating system

None

**Comment by Nana-EC** [comment]
Team to verify resolution

**Comment by MrValioBg** [comment]
The issue as described:
"Steps to reproduce: Execute a contract call that causes a CONTRACT_REVERT_EXECUTED response code." is very likely to be invalid, as this is an old issue from October, where mainnet was still running mono. And it's very possible that this is fixed.

However, the latest example of a similar problem that occurred is from 30.06.2024 - [hashscan txn link.](https://hashscan.io/mainnet/transaction/1719700934.305077051)

This seems like a special case with MAX_CHILD_RECORDS_EXCEEDED, since for other transactions that have failed, the child resultData is correct.

Reproduced the MAX_CHILD_RECORDS_EXCEEDED with IERC20(token).transfer(address, amount) operations as it was in the provided hashscan transaction, as well as with HederaTokenService.transferToken(token, sender, receiver, amount); and inner contract creations. Also, tried running it again in both develop and release 51, but in every case, the produced ContractActions contained resultData, either error or output.

So for now I am marking this as blocked.
Any thoughts? @hashgraph/hedera-smart-contracts-core

**Comment by MrValioBg** [comment]
Update:
After re-checking with the mirror-node team, we confirmed that:
1. Recent transactions that result in throwing the RESULTDATA_NOT_SET error seem to occur only for MAX_CHILD_RECORDS_EXCEEDED.
2. The issue hasn't occurred after 2024-06-29, which is just before mainnet switched to release 51, indicating it was indeed fixed.

So, we decided we could mark this as blocked and revisit it after 2-3 weeks to verify that the issue isn't present.
